### PR TITLE
feat(electron): support console messages and menu items

### LINKED ISSUE #2632: [Feature] Can we have a good example showing how to work with the Electron menu.
Hi.
I'm giving a try to PlayWright to drive my Electron application in my E2E tests.
I'm trying hard to find a way to handle the menu of my Electron application to open what should be a configuration popup window. But actually I can't manage to access and click the menu item to eventually access the configuration dialog window this menu is opening.

I've try something like this...
```
    it('should have access to the menu', async () => {
      await appElectron.evaluate( async ({Menu}) => {
        const menu = Menu.getApplicationMenu();
        console.log('menu items count:', menu.items.length);
        const configMenu = menu.items[6].click();
        configMenu.click();
        const userConfigMenu = configMenu.getMenuItemById('User Config');
        userConfigMenu.click();
      });

      const popup = await appElectron.evaluateHandle((async ({BrowserWindow}) => {
        return BrowserWindow.getFocusedWindow();
      }));

      const title = await popup.evaluateHandle((configPopup) => {
        return configPopup.getTitle();
      });
      console.log('Popup title:', title);
      expect(title).toBeTruthy();
    });
```
... but this doesn't work.

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes https://github.com/microsoft/playwright/issues/2632

### CODE REVIEW

**Review by aslushnikov — APPROVED** [review]

File: src/server/electron.ts [hunk 1]
```diff
@@ -56,6 +59,32 @@ interface ElectronPage extends Page {
...
 
+export class ElectronMenuItem {
+  private _handle: js.JSHandle<MenuItem>;
+
+  constructor(handle: js.JSHandle<MenuItem>) {
+    this._handle = handle;
+  }
+
+  async click() {
+    this._handle.evaluate(menu => menu.click());
```
Comment (dgozman) [inline src/server/electron.ts:70]: Missing await.

File: src/server/electron.ts [hunk 2]
```diff
@@ -161,6 +191,33 @@ export class ElectronApplication extends EventEmitter {
   async evaluateHandle<R, Arg>(pageFunction: types.FuncOn<any, Arg, R>, arg: Arg): Promise<types.SmartHandle<R>> {
     return this._nodeElectronHandle!.evaluateHandle(pageFunction, arg);
   }
+
+  async _onConsoleAPI(event: Protocol.Runtime.consoleAPICalledPayload) {
+    const args = event.args.map(arg => this._nodeExecutionContext!.createHandle(arg));
+    const message = new ConsoleMessage(event.type, undefined, args, toConsoleMessageLocation(event.stackTrace));
+    if (!this.listenerCount(Events.Page.Console))
```
Comment (dgozman) [inline src/server/electron.ts:198]: Events.ElectronApplication.Console

**Review by dgozman — APPROVED** [review]

File: src/server/electron.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/server/electron.ts:70]: yay!

File: src/server/electron.ts (hunk 2 above)
Comment (pavelfeldman) [reply src/server/electron.ts:198]: yay!

**Comment by patgiass** [comment]
What is blocking this PR? Looks like the failing tests are not directly related to the menu.

**Comment by aslushnikov** [comment]
@patgiass windows bot is failing for a good reason

**Comment by ludvigintel** [comment]
Any updates on this one?

**Pull request closed without merging**